game/src/game/player/mod.rs

```diff
@@ -409,6 +419,14 @@ impl WallSlideTime {
     }
 }
 
+/// Indicates that sliding is tracked for this entity
+#[derive(Component, Default, Debug)]
+pub struct WhirlAbility {
+    active: bool,
+    energy: f32,
+    attack_entity: Option<Entity>,
+}
+
```

Comment:
Ended up going for the parallel component approach instead of a state machine, and made it so whirling triggers the Attacking state.
Also integrated attacks with the ability, without needing to many changes.